bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml

```diff
@@ -35,9 +35,292 @@ limitations under the License.
+        </dependency>
+        <dependency>
+          <groupId>commons-collections</groupId>
+          <artifactId>commons-collections</artifactId>
+          <version>3.2.2</version>
+        </dependency>
+
+        <!-- Pull in hbase deps piecemeal and upgrade them to latest versions -->
+
+        <!-- Guava is used throughout the HBase client. Its used for:
+          - connection creation (UserProvider uses Cache & ListeningExecutorService)
+          - value class validation (ie. HColumnDescriptor uses Preconditions)
+          - byte manipulation (TableName.isLegalFullyQualifiedTableName uses Bytes)
+          - etc
+          However, it seems like everything runs with latest version guava
+        -->
+        <dependency>
+          <groupId>com.google.guava</groupId>
+          <artifactId>guava</artifactId>
+          <version>32.1.3-jre</version>
+        </dependency>
+        <!-- HBase 1 uses commons-logging - upgrade it to latest version -->
+        <dependency>
+          <groupId>commons-logging</groupId>
+          <artifactId>commons-logging</artifactId>
+          <version>1.3.0</version>
+        </dependency>
+        <!-- HRegionInfo.createRegionName() uses Hex to encode the md5 hash of the name -->
+        <dependency>
+          <groupId>commons-codec</groupId>
+          <artifactId>commons-codec</artifactId>
+          <version>1.16.0</version>
+        </dependency>
+        <!-- in hbase 1x: org.apache.hadoop.hbase.client.Admin exposes  RpcChannel & BlockingRpcChannel
+        However those methods are not implemented, but to load the Admin class, they must be present.
+        HColumnDescriptor exposes a number of generated proto messages
+        -->
+        <dependency>
+          <groupId>com.google.protobuf</groupId>
+          <artifactId>protobuf-java</artifactId>
+          <version>3.25.1</version>
+        </dependency>
+
+        <dependency>
+          <groupId>com.google.cloud.bigtable</groupId>
+          <artifactId>bigtable-hbase-1.x-shaded</artifactId>
+          <version>2.12.1-SNAPSHOT</version> <!-- {x-version-update:bigtable-client-parent:current} -->
+          <exclusions>
+            <exclusion>
+              <groupId>org.apache.hbase</groupId>
+              <artifactId>hbase-shaded-client</artifactId>
+            </exclusion>
+
+            <!-- This is only necessary modules built in the java-bigtable-hbase repo. End user
+            applications don't need these exclusions.
+            Workaround MNG-5899 & MSHADE-206. Maven >= 3.3.0 doesn't use the dependency reduced
+            pom.xml files when invoking the build from a parent project. So we have to manually exclude
+            the dependencies. Note that this works in conjunction with the manually promoted dependencies
+            in bigtable-hbase-1.x-shaded/pom.xml -->
+            <exclusion>
+              <groupId>com.google.cloud.bigtable</groupId>
+              <artifactId>bigtable-hbase-1.x</artifactId>
```

Comment:
the name of this mode is cbt-shaded, should this be the shaded client?

Reply:
no, it needs both :(

The -shaded one needs to be excluded for the reasons you are thinking (because it is a legitimate dependency). This is done on L293

The non-shaded one (this one) needs to be excluded because maven models are immutable and the shade plugin cant remove the shaded dep. In other words  `bigtable-hbase-1.x-shaded` depends on `bigtable-hbase-1.x`. During packaging, `bigtable-hbase-1.x` gets absorbed into `bigtable-hbase-1.x-shaded` however, the shade plugin can't remove it as a dependency inside the reactor. So within the reactor, another module (like this) when it depends on `bigtable-hbase-1.x-shaded` will get both the absorbed bytecode and the original dep on `bigtable-hbase-1.x`. 

This is only problem within the reactor, and end users dont need to worry about this.  (see the comment above about MNG-5899 & MSHADE-206)